backend/api/activity/index.js

```diff
@@ -6,68 +6,224 @@ const activity = express.Router();
+	  }
+	  
+	  //Delete it
+	  await Activity.deleteOne(getobj);
+	  
+	  //Response
+	  res.sendStatus(200);
+    } catch (error) {
+	  //Error Response
+      res.status(500).send(error);
+    }
   });
 
 /**
- * Searches posts by criteria specified in post data
- * TODO: Add search criteria
- * @body {Number} numPosts - The number of posts to return
+ * Searches activites by custom criteria
+ * @body {Integer} page - The page to send back (Items sent are based on page and pageSize)
+ * @body {Integer} pageSize - The number of activities to return
+ * @body {Integer} sort - The sorting method (see below)
+ * @body {Integer} reverse - Reverses the order of the sort
+ * @body {Integer} omitEnded - Omits activities that have already ended. Defaults to 0.
+ * @body {Integer} omitStarted - Omits activities that have already started. Defaults to 0.
+ *
+ * Sorting methods:
+ * 0 - Start time, soonest first
+ * 1 - Date created, newest first
  */
 activity.route('/search')
   .get(async (req, res) => {
-    const numPosts = req.body.numPosts || 20;
-    try {
-      const getobj = await Activity
-        .find({ creationDateTime: { $exists: true } })
-        .sort({ creationDateTime: -1 })
-        .limit(numPosts);
-      const statusCode = 200;
+	//Default variables for search
+	const page = req.body.page || 1;
+    const pageSize = req.body.pageSize || 20;
+	const sort = req.body.sort || 0;
+	const reverse = req.body.reverse || 0;
+	const omitEnded = req.body.omitEnded || 0;
+	const omitStarted = req.body.omitStarted || 0;
+	const currentDateTime = Date.now();
+	const statusCode = 200;
+	
+	//Handle reversed data
+	var rev = 1;
+	if (reverse) {
+		rev = -1;
+	}
+	
+	//Limit allowed search size
+	if (pageSize > 1000) {
+		return res.status(400).json({
+          error: 'pageSize is too large. Maximum is 1000',
+        });
+	}
+	if (pageSize < 1) {
+		return res.status(400).json({
+          error: 'pageSize must be 1 or greater',
+        });
+	}
+	
+	//Limit page number
+	if (page < 1) {
+		return res.status(400).json({
+          error: 'page must be 1 or greater',
+        });
+	}
+	
+	//Add filter and sort settings to the database query
+	var filterSettings = {}
+	var sortSettings = {}
+	//Sorting methods
+	if (sort == 0) {
+		sortSettings = { startDateTime: (1*rev), _id: (1*rev) }
+	}
+	else if (sort == 1) {
+		sortSettings = { creationDateTime: (-1*rev), _id: (1*rev) }
+	}
+	else {
+	  return res.status(400).json({
+        error: 'Invalid sort method',
+      });
+	}
+	//Filter ended activities
+	if (omitEnded) {
+		filterSettings.endDateTime = { $gt: currentDateTime }
+	}
+	//Filter started activities
+	if (omitStarted) {
+		filterSettings.startDateTime = { $gt: currentDateTime }
+	}
+	
+	//Perform the search
+	try {
+	  const getobj = await Activity
+      .find(filterSettings)
+      .sort(sortSettings)
+      .skip(pageSize * (page-1))
+	  .limit(pageSize);
       res.status(statusCode).json({
         activity: getobj,
       });
```

Comment:
Looks like you're missing a matching catch statement for the find.

Reply:
The try statement you have highlighted there has a catch statement below it. Can you be more specific?

Reply:
Sorry, my bad. It got hidden in the changes.